cool-core/src/main/java/com/nus/cool/core/cohort/refactor/CohortProcessor.java

```diff
@@ -3,60 +3,67 @@
 import java.io.File;
 import java.io.IOException;
 import java.time.LocalDateTime;
-import java.util.HashMap;
-import java.util.HashSet;
+import java.util.*;
 
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.nus.cool.core.cohort.refactor.ageSelect.AgeSelection;
 import com.nus.cool.core.cohort.refactor.birthSelect.BirthSelection;
+import com.nus.cool.core.cohort.refactor.birthSelect.EventSelection;
+import com.nus.cool.core.cohort.refactor.birthSelect.EventSelectionLayout;
 import com.nus.cool.core.cohort.refactor.cohortSelect.CohortSelector;
+import com.nus.cool.core.cohort.refactor.filter.FilterLayout;
+import com.nus.cool.core.cohort.refactor.filter.FilterType;
+import com.nus.cool.core.cohort.refactor.filter.RangeFilter;
 import com.nus.cool.core.cohort.refactor.storage.CohortRet;
 import com.nus.cool.core.cohort.refactor.storage.ProjectedTuple;
 import com.nus.cool.core.cohort.refactor.storage.RetUnit;
+import com.nus.cool.core.cohort.refactor.storage.Scope;
 import com.nus.cool.core.cohort.refactor.utils.DateUtils;
 import com.nus.cool.core.cohort.refactor.valueSelect.ValueSelection;
-import com.nus.cool.core.io.readstore.ChunkRS;
-import com.nus.cool.core.io.readstore.CubeRS;
-import com.nus.cool.core.io.readstore.CubletRS;
-import com.nus.cool.core.io.readstore.HashMetaFieldRS;
-import com.nus.cool.core.io.readstore.MetaChunkRS;
-import com.nus.cool.core.io.readstore.MetaFieldRS;
+import com.nus.cool.core.cohort.refactor.valueSelect.ValueSelectionLayout;
+import com.nus.cool.core.io.readstore.*;
 import com.nus.cool.core.schema.FieldSchema;
 import com.nus.cool.core.schema.FieldType;
 import com.nus.cool.core.schema.TableSchema;
 
 import lombok.Getter;
+import com.nus.cool.core.cohort.refactor.filter.Filter;
 
 public class CohortProcessor {
 
-    private AgeSelection ageSelector;
+    private final AgeSelection ageSelector;
 
-    private ValueSelection valueSelector;
+    private final ValueSelection valueSelector;
 
-    private CohortSelector cohortSelector;
+    private final CohortSelector cohortSelector;
 
-    private BirthSelection birthSelector;
+    private final BirthSelection birthSelector;
 
     @Getter
-    private String dataSource;
+    private final String dataSource;
 
     private ProjectedTuple tuple;
 
     @Getter
-    private CohortRet result;
+    private final CohortRet result;
 
     private String UserIdSchema;
 
     private String ActionTimeSchema;
     
-    private HashSet<String> projectedSchemaSet;
+    private final HashSet<String> projectedSchemaSet;
+
+    private final CohortQueryLayout layout;
 
     public CohortProcessor(CohortQueryLayout layout){
+        this.layout = layout;
```

Comment:
layout won't be used in future. Seperate the IO layer and logic layer
The memory of layout is allowed to be reclaimed.